### PR TITLE
nanos: reduce history screens to 4

### LINKED ISSUE #44: nanos: reduce screen history size
Reduce nanos screen history size to enable u2f endpoint to be integrated, with tz2/tz3 working on nanos.

# Context

Introducing u2f causes the tz2/tz3 signing to crash on the Nanos, due to running out of RAM.

### PR DESCRIPTION
**Pull request by emturner**
fixes #44

### CODE REVIEW

File: app/src/apdu_sign.c [hunk 1]
```diff
@@ -270,8 +324,6 @@ handle_first_apdu(command_t *cmd)
     TZ_CHECK(check_derivation_type(global.path_with_curve.derivation_type));
     TZ_CHECK(cx_blake2b_init_no_throw(&global.apdu.hash.state,
                                       SIGN_HASH_SIZE * 8));
-    tz_ui_stream_init(stream_cb);
```
[the remark this answers is not in the document]
Comment (emturner) [reply app/src/apdu_sign.c:273]: It's not placed anywhere. The current STAX blindsigning workflow does not use `tz_ui_stream` at all

[the remark this answers is not in the document]
Comment (emturner) [reply app/src/apdu_sign.c:273]: Raised #49 to address this, as will also lead to code simplification

**Comment by spalmer25** [comment]
Reducing the number of history screens to 4 has an unexpected effect:
The "Cannot press left" icon appears in the menu when the `Blind Sign` option is activated.

**Review by spalmer25 — COMMENTED** [review]
Reducing the number of history screens to 4 has an unexpected effect:
The "Cannot press left" icon appears in the menu when the `Blind Sign` option is activated

**Review by spalmer25 — CHANGES_REQUESTED** [review]
Reducing the number of history screens to 4 has an unexpected effect:
The "Cannot press left" icon appears in the menu when the `Blind Sign` option is activated.
Do we change that now or leave it for later?
Something like this should fix it:
```c
// In ui_stream.c line 358
-    if (s->current > 0)
-        init[1].text = (const char *)&C_icon_go_left;
-
-    /* Unless we can't... */
-    if (s->current == s->total - TZ_UI_STREAM_HISTORY_SCREENS + 1)
-        init[1].text = (const char *)&C_icon_go_forbid;

+    if (s->current > 0) {
+        /* Unless we can't... */
+        if (s->current == s->total - TZ_UI_STREAM_HISTORY_SCREENS + 1)
+            init[1].text = (const char *)&C_icon_go_forbid;
+        else
+            init[1].text = (const char *)&C_icon_go_left;
+    }
```

**Review by spalmer25 — APPROVED** [review]

**Pull request merged**